crates/astria-gossipnet/src/network.rs

```diff
@@ -85,33 +92,31 @@ impl NetworkBuilder {
         self
     }
 
-    pub fn build(self) -> Result<Network> {
-        Network::new(self.bootnodes, self.port)
+    pub fn keypair(mut self, keypair: Keypair) -> Self {
+        // TODO: load/store keypair from disk
+        self.keypair = Some(keypair);
+        self
     }
-}
 
-impl Default for NetworkBuilder {
-    fn default() -> Self {
-        Self::new()
+    pub fn with_mdns(mut self, with_mdns: bool) -> Self {
+        self.with_mdns = with_mdns;
+        self
     }
-}
 
-pub struct Network {
-    pub multiaddr: Multiaddr,
-    swarm: Swarm<GossipnetBehaviour>,
-    terminated: bool,
-}
+    pub fn with_kademlia(mut self, with_kademlia: bool) -> Self {
```

Comment:
nit: as above with `mdns`.

not-a-nit: please add a doc comment on what this function is doing.